tokio/src/io/util/mem.rs

```diff
@@ -0,0 +1,221 @@
+//! In-process memory IO types.
+
+use crate::io::{AsyncRead, AsyncWrite};
+
+use bytes::{Buf, BytesMut};
+use std::{
+    pin::Pin,
+    sync::{Arc, Mutex},
```

Comment:
Shouldn't this be using the mutex in the loom shim such that it switches to parking_lot if that features is enabled?